### PR TITLE
Support for Ustream, Twitch.tv and Livestream.com

### PR DESCRIPTION
**Pull request by legionaryu**
Probably this is not the best way to do this, but it was the only way that I could get working.

### CODE REVIEW

File: extension/keysocket-ustream.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+window.extisplaing = true;
+
+function clean_injection () {
```
Comment (borismus) [inline extension/keysocket-ustream.js]: I have no idea what this code is doing. Please comment thoroughly.

**Comment by legionaryu** [comment]
In ustream.tv webpage there is an object that controls the player, this object cannot be called by a extension content script, so I inject a script in the DOM to call the play/pause funcions of this object.

Added the comments to explain the basic working mechanism.

**Comment by legionaryu** [comment]
In ustream.tv webpage there is an object that controls the player, this object cannot be called by a extension content script, so I inject a script in the DOM to call the play/pause funcions of this object.

Added the comments to explain the basic working mechanism.

**Comment by legionaryu** [comment]
In ustream.tv webpage there is an object that controls the player, this object cannot be called by a extension content script, so I inject a script in the DOM to call the play/pause funcions of this object.

Added the comments to explain the basic working mechanism.

**Comment by legionaryu** [comment]
I tested right now and it's not working anymore.

**Comment by legionaryu** [comment]
I tested right now and it's not working anymore.

**Comment by legionaryu** [comment]
I tested right now and it's not working anymore.

**Comment by legionaryu** [comment]
I tested right now and it's not working anymore.

**Comment by legionaryu** [comment]
Twitch.tv uses the player api described in https://github.com/justintv/Twitch-API/blob/master/player.md

**Comment by borismus** [comment]
Is this working?

**Comment by legionaryu** [comment]
Yes, it is.

**Comment by legionaryu** [comment]
Livestream.com support using the player api (http://original.livestream.com/userguide/index.php?title=Main_Page&title=API_for_HTML_Developers)

**Pull request merged**